packages/kit/src/runtime/client/client.js

```diff
@@ -269,7 +269,7 @@ export function create_client(app, target) {
 
 	/** @param {import('./types').NavigationFinished} result */
 	function initialize(result) {
-		if (DEV && document.querySelector('vite-error-overlay')) return;
+		if (DEV && result.state.error && document.querySelector('vite-error-overlay')) return;
```

Comment:
> There can also be errors when you're not on the error page

This function only runs after navigations or on the initial load and in those cases you would always be on the error page if there is an error (I think).

Since our tests are concurrent, the overlay is created even if the error was meant for another tab and skipping hydration prevents the rest of the test from running.

It's not the ideal solution, but currently Playwright doesn't offer a way to run specific tests without concurrency.

Right now there is one test where the overlay is expected but the overlay causes flakiness in other tests because it is unintentionally broadcasted to them.

Reply:
Ah that makes sense!